src/gg.regression.unity.bots/Runtime/Scripts/RGSettings.cs

```diff
@@ -69,6 +70,7 @@ public static RGSettings GetOrCreateSettings()
                 _settings.logLevel = DebugLogLevel.Info;
 
                 _settings.feature_StateRecordingAndReplay = true;
+                _settings.feature_CodeCoverage = false;
```

Comment:
My reason for disabling this by default is that it inserts a lot of instrumentation into the assemblies (a call to `RGCodeCoverage.Visit` at every program statement). However, this is invisible to the user (shouldn't affect line numbers in the debug logs either) and is unlikely to have any effect on performance when state recording is disabled, so I could see an argument either way.

Reply:
Yeah I was mainly wondering in the context of performance. You said this doesn't effect much when recording is disabled, but when I am recording, what are the performance implications? Both from the perspective of the runtime and the storage of the results?

Reply:
It would add a small constant performance overhead to every program statement for recording the code coverage. On my machine in Boss Room, I measured it is about 40 nanoseconds overhead per statement. Ultimately the effect this has on the game depends on how much logic they have running per frame. e.g. if the game has a lot of low-level loops that squeeze as much as they can out of the CPU every frame that could stack up. On the other hand, if the game is only implementing high-level logic in its code then the difference is negligible. I imagine most games probably would fall into the latter category.

Really my main hesitance with making this a default option is the huge amount of changes being made to the game assemblies as a result of the code coverage instrumentation. At least with the legacy input instrumentation we are only changing a few method calls, but this inserts a method call into every point in the game program. So, this aspect makes me think this should be an "opt-in" feature, especially since we aren't using this for any user-facing features yet.

Reply:
I agree with the logic about leaving it off by default after reading your descriptions.  I wonder how we could make these details more obvious to developers using the SDK that may choose to enable this option

Reply:
@RG-nAmKcAz what do you think about embedding a short message in the settings UI?

![image](https://github.com/user-attachments/assets/3a3a6ceb-6838-4ba2-a90f-97ceeff5aa10)

Also, I realize we should probably add a warning if the user attempts to create a standalone build with the code coverage enabled, so I will add that shortly as well.

Reply:
what about `This may impact the performance of the game when using the state recording feature.`

Reply:
Added the above UI with the recommended wording. This also made me realize that we should disable the code coverage instrumentation when creating standalone builds in production mode, which is now in as well.

Reply:
@vontell I realize I didn't address the question about storage of the results: the storage cost is a fixed cost for the single metadata file (about 2.5 MB for Boss Room), and then a small cost for storing the code point IDs in each state file. This is just a set of numbers, which for Boss Room takes up roughly ~2 KB of space of the state file.

The sample metadata file is linked in the PR description above, and I'll just paste a snippet here of what the code coverage info looks like in the state recording:

```
"codeCoverage":{
"apiVersion":12,
"coverageSinceLastTick":{
"Unity.BossRoom.Infrastructure":[441,442,447,448,449,450,373,20,21,22,23,24],
"Unity.BossRoom.Utils":[297,298,299,300,301,302,303,304,305,317,322,323,324,325,331,32,33,34,38,35,36,39],
"Unity.BossRoom.Gameplay":[618,619,620,2192,2193,2194,5394,5395,5396,5399,5400,5411,2562,2563,2564,2565,2566,2728,2729,2925,2926,2988,2989,2990,2908,2726,2998,2992,2993,2994,2927,2928,2727,2932,2936,2942,2943,2944,2946,2947,2948,2953,2955,2957,2351,2352,2444,2445,2446,2447,2448,2450,2451,2452,2455,2456,2457,2458,2459,2460,2462,2464,2470,2466,1677,1678,1679,1682,1683,1684,1685,1686,1687,1717,1718,1719,1720,1721,1722,1723,1724,1725,1726,1727,1709,1710,1711,1712,1713,1714,1715,1716,1728,1729,1688,2467,2468,2469,1731,1732,2290,1733,1734,1735,1736,1737,1738,1739,1740,1741,1742,1743,1744,1701,1702,1703,1704,1705,1706,1707,1708,1745,1746,1753,2471,2353,2346,2347,2348,2349,2350,2354,2355,2356,2366,2358,2359,2360,2361,2362,2368,2374,2381,2384,2385,2386,2387,2388,2399,2400,2401,2402,2403,2417,2418,2419,2420,2421,2492,2493,2494,2495,2497,2498,2499,2500,2502,2503,2504,2505,2508,2511,2512,2514,2515,2516,2517,2520,2521,2522,2536,1633,1634,1635,1636,1637,2537,2538,2546,2547,2549,2550,1689,1690,1692,1693,1694,1695,1696,2551,2552,2422,2423,2424,2443,2382,2383,2958,2959,2960,2961,2962,2972,2964,2965,2970,2971,2973,2974,3001,3002,3003,3004,3006,3007,3008,3011,3013,2472,2473,2474,2475,2476,2543,1629,1630,1631,1632,2544,2477,2478,3015,3026,3027,3028,3029,3031,3032,3033,3034,3035,3036,3501,3037,3038,3039,3041,3016,3017,2975,2730,2567,2568],
"Unity.BossRoom.VisualEffects":[22,23,24,25,16,17,18,19,20,21,26,85,86,87],
"Unity.BossRoom.ConnectionManagement":[],
"Unity.BossRoom.UnityServices":[],
"Unity.BossRoom.Audio":[]
}
}
```